src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -651,12 +653,64 @@ public void testGetColumn() throws SQLException {
             getColumnsDWColumns.put(27, "SS_XML_SCHEMACOLLECTION_SCHEMA_NAME");
             getColumnsDWColumns.put(28, "SS_XML_SCHEMACOLLECTION_NAME");
 
+            Map<String, Object> firstRow = new HashMap<String, Object>();
+            String dbName = ds.getDatabaseName();
+            if (null == dbName) {
+                firstRow.put("TABLE_CAT", "master");
+            } else {
+                firstRow.put("TABLE_CAT", ds.getDatabaseName());
+            }
+            firstRow.put("TABLE_SCHEM", "dbo");
+            firstRow.put("TABLE_NAME", tableName);
+            firstRow.put("COLUMN_NAME", "col_1");
+            firstRow.put("DATA_TYPE", 4);
+            firstRow.put("TYPE_NAME", "int");
+            firstRow.put("COLUMN_SIZE", 10);
+            firstRow.put("BUFFER_LENGTH", 4);
+            firstRow.put("DECIMAL_DIGITS", 0);
+            firstRow.put("NUM_PREC_RADIX", 10);
+            firstRow.put("NULLABLE", 0);
+            firstRow.put("REMARKS", null);
+            firstRow.put("COLUMN_DEF", null);
+            firstRow.put("SQL_DATA_TYPE", 4);
+            firstRow.put("SQL_DATETIME_SUB", null);
+            firstRow.put("CHAR_OCTET_LENGTH", null);
+            firstRow.put("ORDINAL_POSITION", 1);
+            firstRow.put("IS_NULLABLE", "NO");
+            firstRow.put("SCOPE_CATALOG", null);
+            firstRow.put("SCOPE_SCHEMA", null);
+            firstRow.put("SCOPE_TABLE", null);
+            firstRow.put("SOURCE_DATA_TYPE", 56);
+            firstRow.put("IS_AUTOINCREMENT", "NO");
+            firstRow.put("IS_GENERATEDCOLUMN", "NO");
+            firstRow.put("SS_IS_SPARSE", 0);
+            firstRow.put("SS_IS_COLUMN_SET", 0);
+            firstRow.put("SS_UDT_CATALOG_NAME", null);
+            firstRow.put("SS_UDT_SCHEMA_NAME", null);
+            firstRow.put("SS_UDT_ASSEMBLY_TYPE_NAME", null);
+            firstRow.put("SS_XML_SCHEMACOLLECTION_CATALOG_NAME", null);
+            firstRow.put("SS_XML_SCHEMACOLLECTION_SCHEMA_NAME", null);
+            firstRow.put("SS_XML_SCHEMACOLLECTION_NAME", null);
+
             try (ResultSet resultSet = databaseMetaData.getColumns(null, null, tableName, "%");) {
                 ResultSetMetaData rsmd = resultSet.getMetaData();
                 int rowCount = 0;
                 while (resultSet.next()) {
                     for (int i = 1; i < rsmd.getColumnCount(); i++) {
-                        assertEquals(rsmd.getColumnName(i), getColumnsDWColumns.values().toArray()[i - 1]);
+                        String columnName = rsmd.getColumnName(i);
+                        Object value = resultSet.getObject(columnName);
+                        if (0 == rowCount) {
+                            Object expectedValue = firstRow.get(columnName);
+                            if (expectedValue instanceof String) {
+                                assertEquals(firstRow.get(columnName).toString().toLowerCase(),
```

Comment:
The fix is good (as confirmed by the customer), but this test isn't accurately testing the changes made by this PR. This new test will pass with or without the changes made in this PR, because when it's time to test the values of the `IS_AUTOINCREMENT` entry, the filter will change the `firstRow.get(columnName).toString()` from `0` to `NO`. To ensure that we've addressed the issue (so that we don't change these values in the future, for example), I think we should additionally check to make sure that `rsmd.getColumnType()` method on the two columns should return 12 (varchar) instead of 4 (int). Other than that, the changes look good to me.

Reply:
You are right, updated the test.